src/BoundaryConditions/no_gradient_open_boudnary.jl

```diff
@@ -0,0 +1,59 @@
+using Oceananigans.Operators: ∂xᶜᶜᶜ
+
+"""
+    ZeroGradient
+
+Zero gradient perepndicular velocity boundary condition.
+
+*Given constant spacing*
+```math
+f′(xᵢ) ≈ f′(xᵢ₋₁) + f′′(xᵢ₋₁)(xᵢ₋₁ - xᵢ) + O(Δx²) = f′(xᵢ₋₁) + f′′(xᵢ₋₁)Δx + O(Δx²),
+
+f′(xᵢ₋₁) ≈ (f(xᵢ) - f(xᵢ₋₂)) / 2Δx
+
+f′(xᵢ) ≈ (f(xᵢ) - f(xᵢ₋₂)) / 2Δx + O(Δx) = 0 ∴ f(xᵢ) ≈ f(xᵢ₋₂) + O(Δx)
+
+f′′(xᵢ₋₁) ≈ (f′(xᵢ) - f′(xᵢ₋₂)) / 2Δx = - f′(xᵢ₋₂) / 2Δx ≈ - (f(xᵢ₋₁) - f(xᵢ₋₃)) / (2Δx)²
+
+∴ f(xᵢ) ≈ f(xᵢ₋₂) + (f(xᵢ₋₁) - f(xᵢ₋₃))/2 + O(Δx²)
+```
+"""
+struct ZeroGradient end
```

Comment:
It's a little confusing --- we already have a `Gradient` type, and one could use `Gradient(0)`. What would you call `Gradient(0)`, other than "zero gradient"? So I'm not sure if we actually need a new type here, or if we want to call it somethign different. We also use thing slike `Flux(nothing)` to mean "zero flux" elsewhere. That could also be used here to mean `ZeroGradient`, eg just

```julia
const ZeroGradient = Gradient{Nothing}
```

rather than defining a new type.

Reply:
Gradient boundary conditions are doing something different though as they are setting the gradient at the mid point between the boundary adjacent and halo nodes to have exactly no gradient. Where as for an open boundary we can't do this because we're trying to approximate the very end to have no gradient (which I wrote out in the docstring).

Reply:
Can we use a different word than "gradient" then?